Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/Common.hs

```diff
@@ -1017,3 +1032,91 @@ sendRideEndMessage bk = case bk.tripCategory of
+  case mbPayoutConfig of
+    Just payoutConfig -> do
+      let isConsideredForPayout = maybe False (\referredAt -> referredAt >= riderConfig.payoutReferrealStartDate) person_.referredAt
+      when (isConsideredForPayout && (fromMaybe False isValidRide) && riderConfig.payoutReferralProgram && payoutConfig.isPayoutEnabled) $ do
+        whenJust person_.referredByCustomer $ \referredByCustomerId -> do
+          personStats <- getPersonStats person_.id
+          QPersonStats.updateReferredByEarning (personStats.referredByEarnings + payoutConfig.referredByRewardAmount) person_.id
+
+          referredByPerson <- QP.findById (Id referredByCustomerId) >>= fromMaybeM (PersonNotFound referredByCustomerId)
+          referredByPersonStats <- getPersonStats referredByPerson.id
+          handlePayout person_ payoutConfig.referredByRewardAmount payoutConfig False referredByPersonStats DLP.REFERRED_BY_AWARD
+          handlePayout referredByPerson payoutConfig.referralRewardAmountPerRide payoutConfig True referredByPersonStats DLP.REFERRAL_AWARD_RIDE
+    Nothing -> logTagError "Payout Config Error" $ "PayoutConfig Not Found for cityId: " <> merchantOperatingCityId.getId <> " and category: " <> show vehicleCategory
+  where
+    handlePayout person amount payoutConfig isReferredByPerson referredByPersonStats entity = do
+      case person.payoutVpa of
+        Just vpa -> do
+          Redis.withWaitOnLockRedisWithExpiry (DIP.payoutProcessingLockKey ride.id.getId) 5 5 $ do
+            case isReferredByPerson of
+              True -> QPersonStats.updateReferralEarningsAndValidActivations (referredByPersonStats.referralEarnings + payoutConfig.referralRewardAmountPerRide) (referredByPersonStats.validActivations + 1) person.id
+              False -> QPersonStats.updateReferredByEarningsPayoutStatus (Just DPS.Processing) person.id
+            phoneNo <- mapM decrypt person.mobileNumber
+            emailId <- mapM decrypt person.email
+            uid <- generateGUID
+            let entityName = entity
+                createPayoutOrderReq = Payout.mkCreatePayoutOrderReq uid amount phoneNo emailId person.id.getId payoutConfig.remark person.firstName vpa payoutConfig.orderType
+            logDebug $ "create payoutOrder with riderId: " <> person.id.getId <> " | amount: " <> show amount <> " | orderId: " <> show uid
+            let serviceName = DEMSC.PayoutService PT.Juspay
+                createPayoutOrderCall = TP.createPayoutOrder merchantId merchantOperatingCityId serviceName
+            merchantOperatingCity <- CQMOC.findById merchantOperatingCityId >>= fromMaybeM (MerchantOperatingCityNotFound merchantOperatingCityId.getId)
+            void $ try @_ @SomeException $ Payout.createPayoutService (cast merchantId) (cast person.id) (Just [ride.id.getId]) (Just entityName) (show merchantOperatingCity.city) createPayoutOrderReq createPayoutOrderCall
```

Comment:
handling left?

Reply:
failed order handling will be in next PR